index.js

```diff
@@ -103,13 +104,133 @@ async function closing (obj, cb) {
  */
 function tieredCache (memOpts, redisOpts) {
   return [
-    cacheManager.caching(Object.assign({}, DEFAULT_MEM_CACHE_OPTS, memOpts)),
-    cacheManager.caching(Object.assign({}, DEFAULT_REDIS_CACHE_OPTS, redisOpts))
-  ]
+    cacheManager.caching(Object.assign({}, DEFAULT_MEM_CACHE_OPTS, memOpts)), cacheManager.caching(Object.assign({}, DEFAULT_REDIS_CACHE_OPTS, redisOpts)) ]
```

Comment:
@n1ywb this could be broken up a bit

Reply:
it's a little clunky because I have to be able to get the raw clients out of the cache stores so I can close them

    {...rest}

would help here too but that's still behind a flag or babel or something

Reply:
also I was thinking you could use the default config objects as prototypes and make copies and override/augments properties